src/onboarding/index.js

```diff
@@ -36,5 +35,10 @@ for (const anchor of anchors) {
 
 // Send message when window closes, so we know to uninstall addon skeleton if user has not agreed.
 window.onunload = () => {
-  browser.runtime.sendMessage({msg: "window_closed"});
+  // Check user_joined and send message only on undefined.
```

Comment:
To be honest I don't know what's going on here, why only on undefined? 🤯

(It would be great if this comment explained why you are doing this and not what you're doing, since I can read what from the code but not why). :)

Reply:
Okay, I'll add a comment. Basically if it's undefined then the user has not yet joined the study. If they have joined then in equals true, if they quit then it equals false (for a brief moment, which then triggers uninstall)